File: frontend/src/platform/lib/community.ts
// The community marketplace bridge, shared by every surface that talks to
// the marketplace backend (the /apps hub's Showcase tab). The backend is
// native server-side now (fused_render/community.py behind POST
// /api/community); it reports its own user-facing failures as
// {status:"error", message}, surfaced verbatim here.
import { postJson } from "@platform/lib/api";

export async function runCommunity<T extends { status?: string; message?: string }>(
  params: Record<string, unknown>,
): Promise<T> {
  const r = await postJson<T>("/api/community", params);
  if (r?.status === "error") throw new Error(r.message || "community backend failed");
  return r;
}

File: fused_render/server/routers/community.py
"""POST /api/community — the community marketplace backend (fused_render/
community.py), for the /apps hub's Showcase tab and the explorer preview's
Clone button.

A sync def on purpose: community.main shells out to git, and FastAPI runs
sync endpoints on its threadpool, keeping those subprocess waits off the
event loop. community.main returns {status:"error", message} for its own
user-facing failures, so the endpoint never raises for those — the client
(platform/lib/community.ts) surfaces the message verbatim.
"""
from fastapi import APIRouter, Body

from fused_render import community

router = APIRouter()


@router.post("/api/community")
def api_community(body: dict = Body(...)):
    return community.main(
        action=str(body.get("action") or "catalog"),
        slug=str(body.get("slug") or ""),
        force=bool(body.get("force")),
    )
